### PR TITLE
Feature/wa 531 create accounts

### PR DESCRIPTION
**Pull request by sche**
- Refactor Account: Use TokenID and WalletID for AccountID
- Add AccountsUpdateDomainService with tests

### CODE REVIEW

File: MultisigWallet/MultisigWalletDomainModel/Account/Account.swift [hunk 1]
```diff
@@ -6,26 +6,59 @@ import Foundation
...
     ///
     /// - Parameters:
-    ///   - id: account identifier
+    ///   - tokenID: account token identifier
     ///   - walletID: wallet identifier
     ///   - balance: balance of the account, in smallest token units
-    public init(id: AccountID, walletID: WalletID, balance: TokenInt) {
+    public init(tokenID: TokenID,
+                walletID: WalletID? = nil,
+                balance: TokenInt? = nil) {
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletDomainModel/Account/Account.swift]: these 3 lines can be merged into one line

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+//
+
+import Foundation
+import BigInt
+
+open class AccountUpdateDomainService {
+
+    public init() {}
+
+    // TODO: Should be done once a wallet is created.
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: I think, this todo is unclear and also shouldn't be here in code.

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        addMissingAccountsForWhitelistedTokenItems()
+        updateBalancesForWhitelistedAccounts()
+    }
+
+    private func addMissingAccountsForWhitelistedTokenItems() {
+        guard let wallet = DomainRegistry.walletRepository.selectedWallet() else { return }
+        let allWalletAccountsTokensIds = DomainRegistry.accountRepository.all()
+            .filter { $0.walletID == wallet.id }
+            .map { $0.id.tokenID }
+        let whitelistedItemsTokensIds = DomainRegistry.tokenListItemRepository.all().filter {
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: I like how it was done in line 22 and 23, maybe format it like there,  for consistency

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        }
+    }
+
+    private func updateBalancesForWhitelistedAccounts() {
+        guard let wallet = DomainRegistry.walletRepository.selectedWallet() else { return }
+        let allWalletAccountsIds = DomainRegistry.accountRepository.all()
+            .filter { $0.walletID == wallet.id }
+            .map { $0.id }
+        let whitelistedItemsTokensIds = DomainRegistry.tokenListItemRepository.all().filter {
+            $0.status == .whitelisted }.map { $0.id }
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: lines 25-39 seems repeating of lines 22-25, can we extract a method or add method on `accountRepository` and on `tokenListitemsRepository` to return `allWalletAccountIds` and `whitelistedItemsTokenIds`

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        let whitelistedAccountsIds = allWalletAccountsIds.filter {
+            whitelistedItemsTokensIds.index(of: $0.tokenID) != nil
+        }
+        updateAccountsBalances(whitelistedAccountsIds)
+    }
+
+    // TODO: add implementation
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: I think, this todo is not needed anymore, is it?

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    private func updateAccountsBalances(_ accountIDs: [AccountID]) {
+        // Stub impl
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: comment not needed

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainServiceTests.swift [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        delay()
+        assertOnlyGNOAccountExistsForSelectedWallet()
+    }
+
+    func test_updateAccountsBalances_doesNotRemoveExitingAccounts() {
+        givenEmptyWalletAndTokenItemsWithWhitelistedGNO()
+        DispatchQueue.global().async {
+            self.accountUpdateService.updateAccountsBalances()
+        }
+        delay()
```
Comment (DmitryBespalov) [inline MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainServiceTests.swift]: these 3 lines (59-62) are repeating in many test cases, please extract them in helper function

**Review by DmitryBespalov — COMMENTED** [review]
Good job!

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift (hunk 5 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: Needed. This is stub implementation.

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainServiceTests.swift (hunk 7 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainServiceTests.swift]: ok

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift (hunk 4 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: I think repos should be dummy.

Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: then can you please add helper methods here to remove repeated code?

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift (hunk 6 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: ok

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift (hunk 5 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: will remove the comment

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift (hunk 4 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: ok

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift (hunk 3 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: ok

File: MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift (hunk 2 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/AccountUpdateDomainService.swift]: ok

File: MultisigWallet/MultisigWalletDomainModel/Account/Account.swift (hunk 1 above)
Comment (sche) [reply MultisigWallet/MultisigWalletDomainModel/Account/Account.swift]: ok

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**